diff -Naur base/meson.build mod/meson.build
--- base/meson.build	2026-07-13 16:07:35
+++ mod/meson.build	2026-08-11 11:37:54
@@ -69,8 +69,10 @@
    # Windows is supported as long as winioctl.h is found
 elif host_machine.system() == 'darwin'
     add_project_arguments(cc.get_supported_arguments('-no-cpp-precomp'), language: 'c')
-    cdata.set('DARWIN_DVD_IOCTL', 1)
-    extra_dependencies += dependency('appleframeworks', modules: ['CoreFoundation', 'IOKit'])
+    if cc.has_header('IOKit/storage/IODVDMediaBSDClient.h', args: test_args)
+      cdata.set('DARWIN_DVD_IOCTL', 1)
+      extra_dependencies += dependency('appleframeworks', modules: ['CoreFoundation', 'IOKit'])
+    endif
 elif host_machine.system() == 'sunos' \
      and cc.has_header('sys/scsi/scsi_types.h', args: test_args) \
      and cc.has_header('sys/scsi/impl/uscsi.h', args: test_args)
diff -Naur base/src/device.c mod/src/device.c
--- base/src/device.c	2026-08-11 11:37:54
+++ mod/src/device.c	2026-08-11 11:37:54
@@ -124,6 +124,8 @@
         return 1;
 
     return 0;
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+    return 0;
 #else
     struct stat fileinfo;
     int ret;
diff -Naur base/src/ioctl.c mod/src/ioctl.c
--- base/src/ioctl.c	2026-07-13 16:07:35
+++ mod/src/ioctl.c	2026-08-11 11:37:54
@@ -229,10 +229,11 @@
                         p_buffer, sizeof(p_buffer), &ulDataLen);
 
     *pi_copyright = p_buffer[ 4 ];
-
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+    *pi_copyright = 0;
+    i_ret = 0;
 #else
 #   error "DVD ioctls are unavailable on this system"
-
 #endif
     return i_ret;
 }
@@ -359,7 +360,8 @@
     }
 
     memcpy( p_key, p_buffer + 4, DVD_DISCKEY_SIZE );
-
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+// nothing
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -492,6 +494,8 @@
 
     memcpy( p_key, p_buffer + 5, DVD_KEY_SIZE );
 
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+// nothing
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -591,6 +595,8 @@
 
     *pi_agid = p_buffer[ 7 ] >> 6;
 
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+// nothing
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -701,6 +707,8 @@
 
     memcpy( p_challenge, p_buffer + 4, DVD_CHALLENGE_SIZE );
 
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+// nothing
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -813,6 +821,8 @@
 
     *pi_asf = p_buffer[ 7 ] & 1;
 
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+// nothing
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -918,6 +928,8 @@
 
     memcpy( p_key, p_buffer + 4, DVD_KEY_SIZE );
 
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+// nothing
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -1003,6 +1015,9 @@
     i_ret = DosDevIOCtl(i_fd, IOCTL_CDROMDISK, CDROMDISK_EXECMD,
                         &sdc, sizeof(sdc), &ulParamLen,
                         NULL, 0, &ulDataLen);
+
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+// nothing
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -1122,6 +1137,8 @@
     if (i_ret == 0)
         memcpy(p_data_buffer, p_buffer, CPRM_MEDIA_ID_SIZE);
 
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+    i_ret = 0;
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -1271,6 +1288,8 @@
         memcpy(p_mkb_pack, p_buffer + 4, CPRM_MKB_PACK_SIZE);
     }
 
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+    i_ret = 0;
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -1383,6 +1402,8 @@
                          &sdc, sizeof(sdc), &ulParamLen,
                          p_buffer, sizeof(p_buffer), &ulDataLen );
 
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+// nothing
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -1494,6 +1515,8 @@
                          &sdc, sizeof(sdc), &ulParamLen,
                          p_buffer, sizeof(p_buffer), &ulDataLen );
 
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+// nothing
 #else
 #   error "DVD ioctls are unavailable on this system"
 
@@ -1623,6 +1646,8 @@
     *p_mask = p_buffer[ 5 ];
     *p_scheme = p_buffer[ 6 ];
 
+#elif defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+// nothing
 #else
 #   error "DVD ioctls are unavailable on this system"
 
